### PR TITLE
Fix mem leak in UUID evaluation

### PR DESCRIPTION
**Pull request by stark-dev**
Signed-off-by: Mauro Guerrera <mauroguerrera@eaton.com>

### CODE REVIEW

**Review by etnKlendathu — APPROVED** [review]

File: server/src/asset/asset.cc [hunk 1]
```diff
@@ -81,17 +81,16 @@ static std::string generateUUID(
 
         std::string src = ns + manufacturer + model + serial;
         // hash must be zeroed first
-        unsigned char* hash = static_cast<unsigned char*>(calloc(SHA_DIGEST_LENGTH, sizeof(unsigned char)));
-
-        SHA1(reinterpret_cast<const unsigned char *>(src.c_str()), src.length(), hash);
+        std::array<unsigned char, SHA_DIGEST_LENGTH> hash;
```
Comment (perrettecl) [inline server/src/asset/asset.cc:84]: we need to set the 0 value in the array

**Review by perrettecl — CHANGES_REQUESTED** [review]

File: server/src/asset/asset.cc (hunk 1 above)
Comment (stark-dev) [reply server/src/asset/asset.cc:84]: `std::array` should get zero initialized automatically, if I checked well on docs

Comment (stark-dev) [reply server/src/asset/asset.cc:84]: @etnKlendathu could you confirm?

Comment (etnKlendathu) [reply server/src/asset/asset.cc:84]: no, I cant... at least not in your case :)
if you will write std::array<unsigned char, SHA_DIGEST_LENGTH> hash = {};
then array will be initialized by probably 0 (T{})... :)

Comment (stark-dev) [reply server/src/asset/asset.cc:84]: Sorry, I will fix it then

**Pull request merged**